hudi-client/hudi-spark-client/src/main/scala/org/apache/spark/sql/HoodieInternalRowUtils.scala

```diff
@@ -224,29 +224,32 @@ object HoodieInternalRowUtils {
         fieldNameStack.pop()
 
         (fieldUpdater, ordinal, value) => {
-          val prevArrayData = value.asInstanceOf[ArrayData]
-          val prevArray = prevArrayData.toObjectArray(prevElementType)
-
-          val newArrayData = createArrayData(newElementType, prevArrayData.numElements())
-          val elementUpdater = new ArrayDataUpdater(newArrayData)
-
-          var i = 0
-          while (i < prevArray.length) {
-            val element = prevArray(i)
-            if (element == null) {
-              if (!containsNull) {
-                throw new HoodieException(
-                  s"Array value at path '${fieldNameStack.asScala.mkString(".")}' is not allowed to be null")
+          if (value == null) {
+            fieldUpdater.setNullAt(ordinal)
+          } else {
+            val prevArrayData = value.asInstanceOf[ArrayData]
+            val prevArray = prevArrayData.toObjectArray(prevElementType)
+
```

Comment:
Thanks for the contribution, can we add some test cases?

Reply:
Could we fix line 191 instead, that could fix all the data type with null values, covering also the atomic data types.

Reply:
thanks, good catch